src/components/availability-panel/drag-drop-context/index.tsx

```diff
@@ -1,100 +1,80 @@
-import DroppableComponent from './DroppableComponent';
-import classNames from '@/components/availability-panel/drag-drop-context/styles.module.scss';
-import { THOUSAND_MILLI_SECONDS, FOURTEEN_DAYS, SECONDS_IN_A_DAY } from '@/components/constants/date'
+import { FC, useState, useEffect } from "react";
+import { DragDropContext, DropResult } from "react-beautiful-dnd";
+import classNames from "@/components/availability-panel/drag-drop-context/styles.module.scss";
+import { dragDropProps } from "@/interfaces/availabilityPanel.type";
+import { toast, ToastTypes } from "@/helperFunctions/toast";
+import task from "@/interfaces/task.type";
+import fetch from "@/helperFunctions/fetch";
+import { ASSIGNED } from "@/components/constants/task-status";
+import DroppableComponent from "./DroppableComponent";
+import SearchInputComponent from "./searchInputComponent";
 
 type NotFoundErrorProps = {
-  message: string,
+  message: string;
 };
 
 const { SUCCESS, ERROR } = ToastTypes;
 
-const NotFoundError: FC<NotFoundErrorProps> = ({ message = 'Not found' }) => (
+const NotFoundError: FC<NotFoundErrorProps> = ({ message = "Not found" }) => (
   <div className={classNames.emptyArray}>
-    <Image
-      src="/ghost.png"
-      alt="ghost"
-      width={160}
-      height={190}
-    />
-    <span className={classNames.emptyText}>
-      {message}
-    </span>
+    <img src="ghost.png" alt="ghost" />
+    <span className={classNames.emptyText}>{message}</span>
   </div>
 );
 
-export const disableDrag = createContext<string[]>([]);
-
-const DragDropContextWrapper: FC<dragDropProps> = ({
+const DragDropcontext: FC<dragDropProps> = ({
   unAssignedTasks,
   idleMembers,
   refreshData,
 }) => {
-  const [toggleSearch, setToggleSearch] = useState<boolean>(false);
+  const [toogleSearchTask, setToogleSearchTask] = useState<boolean>(false);
+  const [toogleSearchMember, setToogleSearchMember] = useState<boolean>(false);
   const [taskList, setTaskList] = useState<Array<task>>(unAssignedTasks);
   const [memberList, setMemberList] = useState<Array<string>>(idleMembers);
   const [isTaskOnDrag, setIsTaskOnDrag] = useState<boolean>(false);
-  const [draggableIds, setDraggableIds] = useState<string[]>([]);
-
-  const ref = useRef<string[]>([]);
+  const [searchTermMember, setSearchTermMember] = useState<string>("");
+  const [searchTermTask, setSearchTermTask] = useState<string>("");
 
   useEffect(() => {
     setTaskList(unAssignedTasks);
     setMemberList(idleMembers);
   }, [unAssignedTasks, idleMembers]);
 
-  const reorder = (list: Array<task | string>, startIndex: number, endIndex: number) => {
+  const handleChangeTask = (e: any) => {
+    setSearchTermTask(e.target.value as string);
+  };
+
+  const handleChangeMember = (e: any) => {
+    setSearchTermMember(e.target.value as string);
+  };
+
+  const onkeypressedTask = (event: any) => {
+    const code = event.charCode || event.keyCode;
+    if (code == 27) {
```

Comment:
NIT: Could we add a comment what 27 signifies here or create a constant like const ESCAPE=27 or something like that?